src/Text/Pandoc/Readers/HTML.hs

```diff
@@ -782,12 +782,12 @@ pImage = do
   let kvs = concatMap getAtt ["width", "height", "sizes", "srcset"]
   return $ B.imageWith (uid, cls, kvs) (escapeURI url) title (B.text alt)
 
-pSamp :: PandocMonad m => TagParser m Inlines 
-pSamp = try $ do 
-  TagOpen open attr' <- pSatisfy $ tagOpen (=="samp") (const True)
+pCodeWithClass :: PandocMonad m => [(T.Text,String)] -> TagParser m Inlines 
+pCodeWithClass elemToClass = try $ do 
+  TagOpen open attr' <- pSatisfy $ tagOpen (flip elem . fmap fst $ elemToClass) (const True)
   result <- manyTill pAny (pCloses open)
   let (ids,cs,kvs) = mkAttr . toStringAttr $ attr'
-  return . B.codeWith (ids,"sample":cs,kvs) .
+  return . B.codeWith (ids,fromJust (lookup open elemToClass) : cs,kvs) .
```

Comment:
Avoid partial functions like `fromJust`.  This will crash if the argument is Nothing.
Even if you're sure the argument won't be Nothing, it's good to code defensively.

Reply:
This `do` block is under a `try` so I think it's okay to throw exception on some failure? I have changed this to `maybe` nonetheless.

Reply:
Parsec `try` isn't what you might expect from `try` in other languages.
If an exception is raised there, the `try` doesn't catch it.

Reply:
Ah I thought it's `try :: Exception e => IO a -> IO (Either e a)`. Thank you.